meson.build

```diff
@@ -0,0 +1,164 @@
+project('ratbagd', 'c',
+	version : '0.2',
+	license : 'MIT/Expat',
+	meson_version : '>= 0.40.0')
+
+ratbagd_version = meson.project_version().split('.')
+
+# Compiler setup
+cc = meson.get_compiler('c')
+cflags = [
+        '-pipe',
+        '-Wall',
+        '-Wextra',
+        '-Wno-inline',
+        '-Wundef',
+        '-Wformat=2',
+	'-Wformat-security',
+	'-Wformat-nonliteral',
+        '-Wlogical-op',
+        '-Wsign-compare',
+        '-Wmissing-include-dirs',
+        '-Wold-style-definition',
+        '-Wpointer-arith',
+        '-Winit-self',
+        '-Wdeclaration-after-statement',
+        '-Wfloat-equal',
+        '-Wsuggest-attribute=noreturn',
+        '-Wmissing-prototypes',
+        '-Wstrict-prototypes',
+        '-Wredundant-decls',
+        '-Wmissing-declarations',
+        '-Wmissing-noreturn',
+        '-Wendif-labels',
+        '-Wstrict-aliasing=2',
+        '-Wwrite-strings',
+        '-Wno-long-long',
+        '-Wno-overlength-strings',
+        '-Wno-unused-parameter',
+        '-Wno-missing-field-initializers',
+        '-Wno-unused-result',
+        '-Werror=overflow',
+        '-Wdate-time',
+        '-Wnested-externs',
+        '-ffast-math',
+        '-fno-common',
+        '-fdiagnostics-show-option',
+        '-fno-strict-aliasing',
+        '-fvisibility=hidden',
+        '-ffunction-sections',
+        '-fdata-sections',
+        '-fstack-protector',
+        '-fstack-protector-strong',
+        '-fPIE',
+        '--param=ssp-buffer-size=4'
+]
+add_project_arguments(cflags, language: 'c')
+
+# Initialize config.h, to be added to in the various options below, config.h
+# is generated at the end of this file
+config_h = configuration_data()
+config_h.set('_GNU_SOURCE', '1')
```

Comment:
I think it's just the direct translation from automake where it was placed into config.h as well, but I'm happy to change this.